### ISSUE TITLE
VDRImageEncoder 

### ISSUE THREAD

**Issue by GuangtaoLyu** [jzhoubu/vsearch#9]
hello, thank you for your great work.
Seems like the proj in VDRImageEncoder isn't being updated? I don't think I saw anywhere where proj would be updated. And I also didn't see where embed(train=True) is being called.

https://github.com/jzhoubu/vsearch/blob/71391b72adc36774b9561c1a8d9063c0444641b9/src/ir/encoder/vdr_crossmodal_image.py#L196

**Comment by jzhoubu** [comment]
Hi, Guangtao. Thank you for your interest. The training code for the multi-modal VDR is not included in this repo. We only provide the inference code here.

**Comment by GuangtaoLyu** [comment]
> Hi, Guangtao. Thank you for your interest. The training code for the multi-modal VDR is not included in this repo. We only provide the inference code here.

ok. thank you.
Can I refer to the implementation below for reproduction?
Secondly, could you provide an estimate of what the top-k value might be?

        outputs = self.bert_model(
            input_ids=input_ids,
            token_type_ids=token_type_ids,
            attention_mask=attention_mask,
        )
        last_hidden_state = outputs.last_hidden_state
        last_hidden_state_ln = self.ln(last_hidden_state)
        vocab_embs = last_hidden_state_ln @ self.bert_model.embeddings.word_embeddings.weight[self.config.shift_vocab_num:, :].t()
        vocab_embs = elu1p(vocab_embs)
        if self.config.pooling == "max":
            vocab_emb = vocab_embs.max(1)[0]
        elif self.config.pooling == "mean":
            if self.config.pooling_topk:
                vocab_emb = vocab_embs.topk(self.config.pooling_topk, dim=1).values.mean(1)
            else:
                vocab_emb = vocab_emb.mean(1)
        else:
            raise NotImplementedError
        vocab_emb = F.normalize(vocab_emb)

**Comment by jzhoubu** [comment]
1. Yes, you can refer to the implementation for reproduction. 
2. For top-k sparsification, we use k=768 for text-to-text VDR and k=512 for multimodal VDR, which match the dimensionality of embeddings from corresponding dense retrieval baseline (e.g., DPR and CLIP).

The main distinction between text-to-text VDR and multi-modal VDR is in the construction of the dataloader. I haven't open-sourced this particular part primarily due to it involving proprietary company implementations and the contributions of others. Using our TextEncoder and ImageEncoder with the same loss function of text-to-text retrieval  + contrastive mask setup should reproduce the multi-modal VDR.
